\documentclass[12pt]{article}

% Essential packages
\usepackage[utf8]{inputenc}
\usepackage[english]{babel}
\usepackage{geometry}
\geometry{a4paper, margin=0.75in}
\usepackage{amsmath, amssymb}
\usepackage{setspace}
\usepackage{enumitem}

% Additional useful packages for academic writing
\usepackage{hyperref}
\usepackage{titlesec}
\usepackage{natbib}  % For bibliography management with author-year citations
\usepackage{graphicx}  % For including figures
\usepackage{enumitem}

% Document formatting
\setlength{\parindent}{0pt}
\setlength{\parskip}{0pt}
\onehalfspacing

% Bibliography settings
\bibliographystyle{abbrv}  % Choose a style: plainnat, unsrtnat, abbrvnat, etc.
% Other common styles: apalike, ieeetr, acm, apa, etc.

% Title information
\title{dCasRx Paper Outline}
\date{\today}

\begin{document}

\maketitle

\section*{Content}
\begin{itemize}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    \item Who would find this work useful and why?
    \begin{itemize}
        \item Researchers in systems and synthetic biology, biomanufacturing, and biological control. The appealing aspect is the ability to quickly train our model with limited data, and easily determine a set of gRNAs to achieve a desired expression profile.
    \end{itemize}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    \item Is this work novel? If so, how?
    \begin{itemize}
        \item Yes, although there are other papers that used \textit{rfx}Cas13d and neural network architecture, there are none that seek to predict gRNA loci effects of nuclease inactive \textit{rfx}Cas13d in bacteria. Previous work includes:

        \begin{itemize}
            \item The use of nuclease active \textit{rfx}Cas13d with other machine learning model (CNNs, Random Forests, Support Vector Classification) to predict gRNA efficacy in mammalian cells \textsuperscript{\citep{Wessels2024, Wei2023, Krohannon2022}}.
            \item This is the first work to use a neural network architecture to predict gRNA loci effects of nuclease inactive \textit{rfx}Cas13d in bacteria that we are aware of.
        \end{itemize}
    \end{itemize}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    \item What portions of the work are the best topics to introduce?
    \begin{itemize}
        \item An illustration of the dCasRx system, with proposed mechanism and plasmid maps. See figure \ref{fig:fig1}.

        \item Representation and results of the GFP tiled spacer library experiments. See figure \ref{fig:fig2}.

        \item Information regarding the experimental data used to trainn and test the model, the model architecture, and the results of the model (specifically the PCA of the embedding space). See figure \ref{fig:fig3}.
            \begin{enumerate}[label=\alph*]
                \item A graphic showing the spacer location relative to the GFP gene, with the corresponding fluorescence intensity for each spacer. See figure \ref{fig:fig3}a.
                \item The time traces of the fluorecsence with the window used for training and testing shown. See figure \ref{fig:fig3}b.
                \item Archectecture of the neural network used to predict the gRNA loci effects. See figure \ref{fig:fig3}c.
                \item The PCA of the embedding space, showing the clustering of the gRNAs based on their location relative to the GFP gene. See figure \ref{fig:fig3}d.
            \end{enumerate}
        \item The results of the Gyrase experiments with emphasiis on how \textit{gyrA} had an increase in growth. See figure \ref{fig:fig4}.
        \item Results of the gyrA RNA seequencing experiments.
        \begin{itemize}
            \item A graphic showing the growth of the dCasRx system with the different gRNAs targeting \textit{gyrA}
            and \textit{gyrB}
        \end{itemize}
    \end{itemize}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    \item Which portions would detract from the quality of the work?
        \begin{itemize}
            \item Possibly the figure comparing the growth deffects of CasRx, dCasRx, and dCas9. This is not the focus of the paper, and may detract from the main message.
        \end{itemize}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    \item How would you characterize the content? Is it conceptual, developmental, analytical, or descriptive?
    \begin{itemize}
        \item The contnet is mainly develeopmental and descriptive, as it describes the development of the dCasRx system and the results of the experiments. However, it also has analytical components, as it includes the analysis of the data collected from the experiments and the machine learning model used to predict gRNA loci effects.
    \end{itemize}
\end{itemize}

\section*{Main Purpose}
To show that deactivate CasRx, coupled with a predicive neutral network model, can be used to create a variety of complex changes to plasmid or chromosoal gene expression in bacteria.
\begin{enumerate}
    \item Show that dCasRx can create diverse changes to plasmid gene exression based sequencial location of the protospacer.
    \item Provide a model archetecture that can predict these effects with a relatively small amount of training data.
    \item Show that similar changes can be made to chromosomal gene expression.
    \item Show that changess in house-keeper genes allow pheneotypics changes. In the case of Gyrase, the changes differ significantly from chemical inhigiotion.
\end{enumerate}

\section*{Abstract}
\begin{itemize}
    \item \textbf{Main research question/problem statement:} Can mRNA tarteging CRISPR-cas systems be used to prodcuce changes in bacterial systems in a predictive manner? If so what, what type of change?
    \item \textbf{Methods used:}
    \begin{itemize}
        \item Annealed Oligo and Golen Gate clonning to assemble gRNA librarys.
        \item Microplate reader based fluorescence and absorbance measurements were used to quantify gene expression and growth rates. Fluorescence was measured to assess the expression levels of GFP, while absorbance at 600 nm (OD600) was used to monitor bacterial growth.
        \item Machine learning techniques, specifically a neural network architecture, were employed to predict the effects of gRNA loci on gene expression.
        \item Analysis of the embedding space using PCA to visualize the clustering of gRNAs based on their location relative to the target gene.
    \end{itemize}
    \item \textbf{Key findings:}
        \begin{itemize}
            \item The dCasRx system can produce diverse changes in plasmid gene expression based on the sequential location of the protospacer.
            \item The neural network model can predict gRNA loci effects with a relatively small amount of training data.
            \item Similar changes can be made to chromosomal gene expression, demonstrating the versatility of the dCasRx system.
            \item Changes in house-keeping genes, such as gyrase, lead to phenotypic changes that differ significantly from chemical inhibition.
        \end{itemize}
    \item \textbf{Significance and implications:} This work demonstrates the potential of dCasRx systems in synthetic biology and biomanufacturing, providing a framework for predictive control of gene expression in bacteria. The findings have implications for the development of advanced genetic engineering tools and strategies for biological control.
\end{itemize}

\section*{Introduction}
\begin{itemize}
    \item Background and context of CRISPR systems \textsuperscript{\citep{abudayyeh2017}}
    \item Current state of knowledge on CasRx \textsuperscript{\citep{konermann2018}}
    \item Gap in knowledge/research problem
    \item Research objectives related to Cas13d functionality \textsuperscript{\citep{yan2018}}
    \item Hypotheses based on prior transcriptional regulation work \textsuperscript{\citep{GILBERT2014647}}
    \item Significance of the study
    \begin{itemize}
        \item This study provides a novel approach to using nuclease-inactive CasRx systems for predictive control of gene expression in bacterial systems.
        \item It bridges the gap between CRISPR-based transcriptional regulation and machine learning, enabling precise and efficient genetic engineering.
        \item The findings have potential applications in synthetic biology, biomanufacturing, and the development of advanced tools for biological control.
        \item By demonstrating the versatility of dCasRx in both plasmid and chromosomal contexts, this work lays the foundation for future research into programmable gene expression systems.
    \end{itemize}
\end{itemize}

\section*{Results}
\begin{itemize}
    \item dCasRx effectors provide a wide variety of chaneges in plasmid gene expression based on the sequential location of the protospacer (\ref{fig:fig1}, \ref{fig:fig2}).
    \item Our neural network model can predict gRNA loci effects with a relatively small amount of training data (\ref{fig:fig3}).
    \item Similar changes can be made to chromosomal gene expression, demonstrating the versatility of the dCasRx system (\ref{fig:fig4}).
    \item These changes, as exemplified by the gyrase example, lead to phenotypic changes which differ significantly from traditional chemical inhibition (\ref{fig:fig5}).
    \item (Do we include this). dCasRx inhibition results in a smaller growth defect when compared to dCas9 and CasRx (\ref{fig:fig6}).
\end{itemize}

\section*{Discussion}
\begin{itemize}
    \item Interpretation of results
    \item Comparison with existing literature
    \item Implications of findings
    \item Limitations of the study
    \item Future research directions
\end{itemize}

\section*{Conclusion}
\begin{itemize}
    \item Summary of key findings
    \item Broader implications
    \item Final thoughts
\end{itemize}

\section*{Methods}
\begin{itemize}
    \item Experimental design
    \item dCasRx system description
    \item gRNA design and validation
    \item Data collection procedures
    \item Analysis techniques
\end{itemize}

% The bibliography command will insert references here
\bibliography{ATCitations}
\citep{Wessels2024} % Example citation to ensure \citation command is used

\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{../../Figures/paperFig1v2_Mech_and_plasmids.png}
  \caption{Mechanism and plasmid maps for the dCasRx system.}
  \label{fig:fig1}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{../../Figures/80memberLibraryDifferenceBarChart_2_0_3.png}
    \caption{Mechanism and plasmid maps for the dCasRx system.}
    \label{fig:fig2}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{../../Figures/paperFigIIIvBlurry.png}
    \caption{Mechanism and plasmid maps for the dCasRx system.}
    \label{fig:fig3}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{../../Figures/GyraseKnockdown2.png}
    \caption{Mechanism and plasmid maps for the dCasRx system.}
\label{fig:fig4}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=\textwidth]{../../Figures/dCas9_CasRxComp.png}
    \caption{Mechanism and plasmid maps for the dCasRx system.}
\label{fig:fig6}
\end{figure}


\section*{Supplementary Materials}
\begin{itemize}
    \item Additional data
    \item Extended methods
    \item Supporting analyses
\end{itemize}

\end{document}